### PR TITLE
Replace ListWatchUntil with UntilWithSync to avoid watch timeouts

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
ListWatchUntil can't handle closed watches and is being replaced by UntilWithSync based on informers.

**Special notes for your reviewer**:
Continued from #72648

**Release note**:
NONE

/cc @liggitt

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/certificate/csr/csr.go [hunk 1]
```diff
@@ -95,7 +98,7 @@ func WaitForCertificate(client certificatesclient.CertificateSigningRequestInter
 				options.FieldSelector = fieldSelector
 				return client.Watch(options)
 			},
-		},
+		}, &v1.Secret{}, nil,
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/util/certificate/csr/csr.go]: this expects `&v1beta1.CertificateSigningRequest{}` objects, not secrets

**Comment by wojtek-t** [comment]
/ok-to-test
/lgtm
/approve

@tedyu - you need to sign the CLA.

**Comment by tedyu** [comment]
event is assigned on line 90 from the return value of watchtools.UntilWithSync()
Not sure why the compiler complained about it.

w.r.t. CLA, I have "You just signed CNCF Individual Contributor License Agreement" in my gmail inbox.

**Comment by tedyu** [comment]
Let me close this and open a new PR to get over CLA warning.

**Pull request closed without merging**